src/main/java/com/redhat/devtools/lsp4ij/LanguageServerWrapper.java

```diff
@@ -635,8 +635,8 @@ public boolean canOperate(Project project) {
      *                 by using a blocking read action.
      * @return the completable future with the language server instance or null.
      */
-    CompletableFuture<@Nullable LanguageServer> connect(@NotNull VirtualFile file,
-                                                        @Nullable Document document) {
+    public CompletableFuture<@Nullable LanguageServer> connect(@NotNull VirtualFile file,
```

Comment:
Had to make this `public` so that I could connect files that have not ever been opened in editor tabs to the respective language servers. See `LSPDiagnosticAnnotator#connectIfNecessary()` for more details.

Reply:
If I understand correctly your code, you need to call connect for file which are not opened, right?

If it that we need to find an another way because connect call LSP didOpen notification which means "the file is opened in an editor".

Reply:
Yeah, this approach is a dead end. I tried to add support for [pull diagnostics](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_pullDiagnostics) instead as that would do exactly what we need, but unfortunately that capability is not very well supported, at least among the language servers that I tried. The TypeScript server doesn't support it, and the CSS server said it did but then threw an exception when it received the request.

I think this overall feature may need to wait until pull diagnostics are more broadly supported.